fades/main.py

```diff
@@ -218,8 +219,11 @@ def go():
         logger.debug("Dependencies from source file: %s", indicated_deps)
         docstring_deps = parsing.parse_docstring(args.child_program)
         logger.debug("Dependencies from docstrings: %s", docstring_deps)
-    reqfile_deps = parsing.parse_reqfile(args.requirement)
-    logger.debug("Dependencies from requirements file: %s", reqfile_deps)
+    reqfile_deps = {}
+    for rfpath in args.requirement:
+        rfdeps = parsing.parse_reqfile(rfpath)
+        logger.debug('Dependencies from requirements file "%s": %s', rfpath, rfdeps)
+        reqfile_deps.update(rfdeps)
```

Comment:
You can not do this, see that `parsing.parse_reqfile` returns a dict whose key is the "repo", so you would be overriding stuff from one req file into the other.

You need to get this info and use it below in the the `_merge_deps`  call.

What we *really* should do here is to extract the "dependencies grabbing" from inside here and put it somewhere else so it can be properly tested... but that is maybe work for another branch...

Reply:
I hope the update to the PR addresses your comment.